add ability to get the list of associated PRs for commit

dmitrykiselev27 opened the pull request:
Use application/vnd.github.groot-preview+json request header to get the list of associated PRs for commit because the endpoint is in preview

omgjlk commented:
Oh and you'll need to add yourself as an author. See https://github.com/sigmavirus24/github3.py/blob/master/CONTRIBUTING.rst

sigmavirus24 commented on src/github3/repos/commit.py:
```diff
@@ -139,6 +144,24 @@ def comments(self, number=-1, etag=None):
         url = self._build_url("comments", base_url=self._api)
         return self._iter(int(number), url, RepoComment, etag=etag)
 
+    def associated_prs(self):
```
I'm pretty sure we always want to ensure the user can pass a number here. Even if we can't pass that to the API, we can provide that functionality and this would be one odd method out.

Further, we prefer full names, so this should be `associated_pull_requests`

dmitrykiselev27 commented:
@omgjlk @sigmavirus24 thanks for the feedback. 
I've added missing unit test, changed method name to `associated_pull_requests`, and added missing parameters. 
Also, the branch is up-to-date with master.

sigmavirus24 commented:
I'm happy to merge this. Would you be willing to fix the issues in CI around linting? 

```
tests/integration/test_repos_repo.py:979:17: E741 ambiguous variable name 'l'
```
